pkg/operator/ceph/cluster/predicate.go

```diff
@@ -131,8 +131,11 @@ func watchControllerPredicate(rookContext *clusterd.Context) predicate.Funcs {
 				}
 				diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
 				if diff != "" {
-					// Set the cancellation flag to stop any ongoing orchestration
-					rookContext.RequestCancelOrchestration.Set()
+					// Set the cancellation flag to stop any ongoing orchestration only if ceph version is updated
+					if objOld.Spec.CephVersion.Image != objNew.Spec.CephVersion.Image {
```

Comment:
Instead of only setting the cancel if the version changed, can we set the cancel only if a reconcile is currently in progress? The issue seems to be that we are setting the cancel when there is no reconcile in progress, which means that the next reconcile will be cancelled when it shouldn't be.

Reply:
How about checking the `[clusterState](https://github.com/rook/rook/blob/162fd1b4420ad5bb3649be74d5054b402212b9c9/pkg/apis/ceph.rook.io/v1/types.go#L430)`?  `Creating`, `Updating` and `Connecting` would imply that there is an ongoing reconcile in progress.